js/form_editor_controller.js

```diff
@@ -1064,6 +1125,15 @@ class GlpiFormEditorController
             extra_data,
         );
 
+        // Update the question type
+        this.#setItemInput(question, "type", type);
+
+        // Convert the default value to match the new type
+        if (
+            this.#converters[old_type] !== undefined && this.#converters[old_type][type] !== undefined) {
+            this.#converters[old_type][type](question, extracted_default_value);
+        }
```

Comment:
It seems that it makes the registration of all know types mandatory. I am not sure it is a maintainable solution. The previous one was probably a better solution.

Reply:
Maybe a solution that ignore question type and just indicate value types would be simpler ?

We could define a few known supported types for export values (raw string, array of strings, dropdown id, array of dropdown ids, ...) and then the question could deal with this known types instead of checking the question type directly.

For example the short answer question type would support receiving a raw string but nothing else.
That doesn't stop an hypothetical plugin from being able to work with it.

Reply:
It is probably better to focus on the value type rather than the question class. However, changing the question type is not something that people will do often. Please try to keep things simple.